test/functional/ui/cmdline_spec.lua

```diff
@@ -187,6 +187,36 @@ describe('external cmdline', function()
 
   end)
 
+  it("works with inputlist()", function()
+    feed(':call inputlist(["input0", "input1"])<cr>')
+    screen:expect([[
+      {1:~                        }|
+      {1:~                        }|
+      input0                   |
+      input1                   |
+            ^                   |
```

Comment:
Hmm cursor position is weird, ideally it should be on the first column, or perhaps avoid the last blank line altogether if possible.

Reply:
I think this code is responsible for putting the last line as well as the cursor position:
```c
// from line 11279 in eval.c
TV_LIST_ITER_CONST(argvars[0].vval.v_list, li, {
    msg_puts(tv_get_string(TV_LIST_ITEM_TV(li)));
    msg_putchar('\n');
  });
```
Removing the newline would be complicated... Also, how do we change the cursor position? It stays at the position that `msg_putchar()` or `msg_puts()` leaves it.

Reply:
I think you can use `TV_LIST_ITEM_NEXT(list, li) != NULL` to check if an item is last and then avoid the `msg_putchar('\n')`. ( when `ui_is_external(kUICmdline)` is true)

Alternatively, i think you can just add an `msg_putchar('\r')` at the end to move it back to first column.

Reply:
Better to check against equality to `tv_list_last()`: it makes the intention more clear.

Reply:
After rebasing with master, if `msg_putchar('\n')` is not present after every list item, the list isn't displayed correctly. Should I revert?